src/gui/src/staDescriptors.cpp

```diff
@@ -0,0 +1,498 @@
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include "staDescriptors.h"
+
+#include <QInputDialog>
+#include <QStringList>
+#include <boost/algorithm/string.hpp>
+#include <iomanip>
+#include <limits>
+#include <queue>
+#include <regex>
+#include <sstream>
+
+#include "db_sta/dbNetwork.hh"
+#include "db_sta/dbSta.hh"
+#include "sta/FuncExpr.hh"
+#include "sta/Liberty.hh"
+#include "sta/PortDirection.hh"
+#include "utl/Logger.h"
+
+namespace gui {
+
+static void capitalize(std::string& str)
+{
+  str[0] = std::toupper(str[0]);
+}
+
+static void add_if_true(Descriptor::Properties& props,
+                        const char* name,
+                        const bool value)
+{
+  if (value) {
+    props.push_back({name, value});
+  }
+}
+
+// T may be MinMax or RiseFall
+template <typename T>
+using LimitFunc = std::function<
+    void(sta::LibertyPort* port, const T* t, float& limit, bool& exists)>;
+
+template <typename T>
+static void add_limit(Descriptor::Properties& props,
+                      sta::LibertyPort* port,
+                      const char* label,
+                      LimitFunc<T> func)
+{
+  for (const auto* elem : T::range()) {
+    bool exists;
+    float limit;
+    func(port, elem, limit, exists);
+    if (exists) {
+      std::string elem_str = elem->asString();
+      capitalize(elem_str);
+      props.push_back({fmt::format("{} {}", elem_str, label), limit});
+    }
+  }
+};
+
+//////////////////////////////////////////////////
+
+LibertyLibraryDescriptor::LibertyLibraryDescriptor(odb::dbDatabase* db,
+                                                   sta::dbSta* sta)
+    : db_(db), sta_(sta)
+{
+}
+
+std::string LibertyLibraryDescriptor::getName(std::any object) const
+{
+  return std::any_cast<sta::LibertyLibrary*>(object)->name();
+}
+
+std::string LibertyLibraryDescriptor::getTypeName() const
+{
+  return "Liberty library";
+}
+
+bool LibertyLibraryDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  return false;
+}
+
+void LibertyLibraryDescriptor::highlight(std::any object,
+                                         Painter& painter) const
+{
+}
+
+Descriptor::Properties LibertyLibraryDescriptor::getProperties(
+    std::any object) const
+{
+  auto library = std::any_cast<sta::LibertyLibrary*>(object);
+
+  Properties props;
+
+  auto gui = Gui::get();
+  SelectionSet cells;
+  sta::LibertyCellIterator cell_iter(library);
+  while (cell_iter.hasNext()) {
+    cells.insert(gui->makeSelected(cell_iter.next()));
+  }
+  props.push_back({"Cells", cells});
```

Comment:
might be nice to have:
* delayModelType
* nominalProcess
* nominalVoltage
* nominalTemperature
* defaultMaxSlew
* defaultMaxFanout
* defaultMaxCapacitance

It would also be nice to know what corner this library is associated with:
https://github.com/The-OpenROAD-Project/OpenSTA/blob/20925bb00965c1199c45aca0318c2baeb4042c5a/include/sta/Corner.hh#L110